shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/impl/ShardingSelectStatementValidator.java

```diff
@@ -48,5 +54,8 @@ public void postValidate(final ShardingRule shardingRule, final SQLStatementCont
         if (!routeContext.isFederated() && needCheckDatabaseInstance) {
             Preconditions.checkState(routeContext.isSingleRouting(), "Sharding value must same with subquery.");
         }
+        if (!sqlStatementContext.getSqlStatement().getUnionSegments().isEmpty() && routeContext.getRouteUnits().size() > 1) {
+            throw new ShardingSphereException("UNION can not support sharding route to multiple data nodes.");
```

Comment:
@tuichenchuxin `SELECT ... UNION can not support sharding route to multiple data nodes.`